src/DynamoCoreWpf/Views/Preview/PreviewControl.xaml.cs

```diff
@@ -435,6 +436,28 @@ private void RefreshExpandedDisplay(Action refreshDisplay)
             );
         }
 
+        string checkNumberFormat(string numberValue)
+        {
+            if (IsDoubleRealNumber(numberValue))
+            {
+                return Convert.ToDouble(numberValue).ToString(nodeViewModel.DynamoViewModel.PreferenceSettings.NumberFormat);
+            }
+            else
+            {
+                return numberValue;
+            }
+        }
+
+        private bool IsDoubleRealNumber(string valueToTest)
```

Comment:
Do we really need this function? Since this is only a one line if check, maybe we can just fold it to the function above?
```
if (double.TryParse(valueToTest, out double d) && !Double.IsNaN(d) && !Double.IsInfinity(d))
            {
                return Convert.ToDouble(numberValue).ToString(nodeViewModel.DynamoViewModel.PreferenceSettings.NumberFormat);
            }
            else
            {
                return numberValue;
            }
```

Reply:
> Do we really need this function? Since this is only a one line if check, maybe we can just fold it to the function above?
> 
> ```
> if (double.TryParse(valueToTest, out double d) && !Double.IsNaN(d) && !Double.IsInfinity(d))
> {
>        return Convert.ToDouble(numberValue).ToString(nodeViewModel.DynamoViewModel.PreferenceSettings.NumberFormat);
> }
> else
> {
>        return numberValue;
>  }
> ```

just wanted to keep separate it, probably it would be part of a helper class or similar so we can move latter, like ADSK.Dynamo.Helpers.IsDoubleRealNumber (maybe we already have a similar) or you  approach to keep it simple is totally valid, please your thoughts.

Reply:
OK, in that case, I am fine with keeping it, but we should then probably move this function to one of DynamoUtilites file, like https://github.com/DynamoDS/Dynamo/blob/master/src/DynamoCoreWpf/Utilities/PreferencesPanelUtilities.cs. Also maybe pick a name easier to understand, like `IsValidDoubleNumber`?